.ci/pytorch/test.sh

```diff
@@ -547,43 +547,60 @@ test_libtorch() {
     ln -sf "$TORCH_LIB_DIR"/libtbb* "$TORCH_BIN_DIR"
     ln -sf "$TORCH_LIB_DIR"/libnvfuser* "$TORCH_BIN_DIR"
 
+    export CPP_TESTS_DIR="${TORCH_BIN_DIR}"
+
     # Start background download
-    python tools/download_mnist.py --quiet -d test/cpp/api/mnist &
+    MNIST_DIR="${PWD}/test/cpp/api/mnist"
+    python tools/download_mnist.py --quiet -d "${MNIST_DIR}" &
 
-    # Make test_reports directory
-    # NB: the ending test_libtorch must match the current function name for the current
-    # test reporting process to function as expected.
-    TEST_REPORTS_DIR=test/test-reports/cpp-unittest/test_libtorch
-    mkdir -p $TEST_REPORTS_DIR
+    # Prepare the model used by test_jit, the model needs to be in the test directory
+    # to get picked up by run_test
+    pushd test
+    python cpp/jit/tests_setup.py setup
+    popd
 
     # Run JIT cpp tests
-    python test/cpp/jit/tests_setup.py setup
-
     if [[ "$BUILD_ENVIRONMENT" == *cuda* ]]; then
-      "$TORCH_BIN_DIR"/test_jit  --gtest_output=xml:$TEST_REPORTS_DIR/test_jit.xml
-      "$TORCH_BIN_DIR"/nvfuser_tests --gtest_output=xml:$TEST_REPORTS_DIR/nvfuser_tests.xml
+      python test/run_test.py --cpp --verbose -i cpp/test_jit cpp/nvfuser_tests
     else
-      "$TORCH_BIN_DIR"/test_jit  --gtest_filter='-*CUDA' --gtest_output=xml:$TEST_REPORTS_DIR/test_jit.xml
+      # CUDA tests have already been skipped when CUDA is not available
+      python test/run_test.py --cpp --verbose -i cpp/test_jit -k "not CUDA"
     fi
 
     # Run Lazy Tensor cpp tests
     if [[ "$BUILD_ENVIRONMENT" == *cuda* && "$TEST_CONFIG" != *nogpu* ]]; then
-      LTC_TS_CUDA=1 "$TORCH_BIN_DIR"/test_lazy  --gtest_output=xml:$TEST_REPORTS_DIR/test_lazy.xml
+      LTC_TS_CUDA=1 python test/run_test.py --cpp --verbose -i cpp/test_lazy
     else
-      "$TORCH_BIN_DIR"/test_lazy  --gtest_output=xml:$TEST_REPORTS_DIR/test_lazy.xml
+      python test/run_test.py --cpp --verbose -i cpp/test_lazy
     fi
 
-    python test/cpp/jit/tests_setup.py shutdown
+    # Cleaning up test artifacts in the test folder
+    pushd test
```

Comment:
I'm curious why the `pushd/popd` was needed now when it wasn't needed before

Reply:
This script `test/cpp/jit/tests_setup.py` creates and cleans up a dummy JIT output for testing to the current directory.  Previously, when the C++ tests was run directly as an executable, there was no need to change the directory. On the other hand, `run_test.py` changes the working directory to `test`, thus I need to do the same here to create that dummy output in `test` folder instead.